GoogleForCreators/web-stories-wp
# Dashboard: Updated animation provider to support amp animations in amp stories

This PR updates our Story Animation component to support `AMP` animations.  It touches a lot of files but mostly because of some component renames that I made (so many of the files are just 1 line changes).

I manually generated an `amp-story` in Storybook to showcase how we'll be able to apply animations using our data structure and provider.  

You can access the demo in Storybook at:
`Dashboard -> Components -> StoryAnimation -> AMP Story`

![amp-story-with-animations](https://user-images.githubusercontent.com/40646372/82273901-8f03e380-9933-11ea-94c5-95f5a8b8a741.gif)

## Testing Instructions
- Go to the "AMP Story" storybook page and see if it you see animations on each page.

**NOTE**:  Sometimes the animation on the fist page does not load, but if you go to the next page and back, then it should bounce in.  I don't believe this is an error on our part, I think it has something to do with loading the amp-story in Storybook and the 'visible' trigger not being activated when it should.  If this happens when we generate and preview a full amp-story using our app, then we might have to investigate further.

## TODO
- This update breaks our existing Storybook Animations (found under `Dashboard -> Animations`), so they'll have to be updated to use the new world order of animation (should be pretty easy to do).
- The appropriate `output` elements still need to be updated so an `amp-story` generated by our editor with animations will animate properly.  That will come in a later PR.
